### PR TITLE
Clarity info from reporting-app rest api

### LINKED ISSUE #118: Migrate query from the LIMS API
in `clarity`, some functions can use the Reporting-App LIMS endpoint instead of `pyclarity-lims`
The Reporting app end point are easier to Mock in integration tests so they should be preferred.

### PR DESCRIPTION
**Pull request by tcezard**
This PR change the behaviour of a few function that now retrieve information from the Reporting App rest API before they get it from `pyclarity-lims`
Closes #118

### CODE REVIEW

File: egcg_core/clarity.py [hunk 1]
```diff
@@ -181,17 +204,22 @@ def get_user_sample_name(sample_name, lenient=False):
...
 
 def get_sample_sex(sample_name):
-    sample = get_sample(sample_name)
-    if sample:
-        return sample.udf.get('Sex') or sample.udf.get('Gender')
+    sex = lims_samples_info(sample_name).get('Sex') or lims_samples_info(sample_name).get('Gender')
+    if not sex:
+        sample = get_sample(sample_name)
+        if sample:
+            sex = sample.udf.get('Sex') or sample.udf.get('Gender')
```
Comment (lukbut) [inline egcg_core/clarity.py:221]: So what happens if sample does not exist in this case, and therefore the if statement is not executed?

Comment (tcezard) [reply egcg_core/clarity.py:221]: `sex` should be set to `None`

**Review by lukbut — APPROVED** [review]

**Review by mwhamgenomics — COMMENTED** [review]
Looks good, but I don't see why we're falling back on pyclarity_lims - it looks for the same data via the Lims XML API, so if the UDF isn't found in `lims/sample_info`, then pyclarity_lims won't find it either. Is this for catching any problems in Reporting-App's Lims endpoints?

**Comment by tcezard** [comment]
> Looks good, but I don't see why we're falling back on pyclarity_lims - it looks for the same data via the Lims XML API, so if the UDF isn't found in `lims/sample_info`, then pyclarity_lims won't find it either. Is this for catching any problems in Reporting-App's Lims endpoints?

That's a fair comment.
I was looking to be backward compatible in case our rest API would not be available although that seems very unlikely.
I don't think it hurts to have it there but I acknowledge that the use will be limited.
Happy to remove the fallback if you have strong feeling about it.

**Comment by mwhamgenomics** [comment]
Nah, no strong feelings here - this can be merged

**Review by mwhamgenomics — APPROVED** [review]

**Pull request merged**